data-propagation.md

```diff
@@ -2,13 +2,16 @@
 
 The filecoin network needs to broadcast blocks and messages to all peers in the network. This document details how that process works.
 
-Both blocks and messages are propagated using the gossipsub libp2p pubsub router. The pubsub messages are authenticated. For blocks, the pubsub hop validation function is set to check that the block is valid before re-propagating. For messages, a similar validity check is run, the signature must be valid, and the account in question must have enough funds to cover the actions specified.
+Both block headers and message references are propagated using the [gossipsub libp2p pubsub router](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub). Every full node must implement and run that protocol. All pubsub messages are authenticated and must be validated before being propagated further.
```

Comment:
s/references//

Whole messages are broadcast through the pubsub layer on the messages channel. the blocks channel broadcasts the block headers and message cids.

bitswap is only used in the cases where we missed pubsub messages (around joining the network, or if hiccups occur, such as a miner mining a message that they didnt propagate)